mlir/include/mlir/Analysis/Presburger/IntegerRelation.h

```diff
@@ -738,6 +738,12 @@ class IntegerRelation {
   /// Same as findSymbolicIntegerLexMin but produces lexmax instead of lexmin
   SymbolicLexOpt findSymbolicIntegerLexMax() const;
 
+  /// Searches for a constraint with a non-zero coefficient at `colIdx` in
+  /// equality (isEq=true) or inequality (isEq=false) constraints.
+  /// Returns true and sets row found in search in `rowIdx`, false otherwise.
+  bool findConstraintWithNonZeroAt(unsigned colIdx, bool isEq,
+                                   unsigned *rowIdx) const;
```

Comment:
I would prefer to return an optional instead.

Reply:
I'm not adding a new method here, but only moving it. (This is an NFC PR.) If we want to change the signature, it should ideally be done in another commit to not mix pure code movement with it.

Reply:
I'll anyway improve the signature as you suggested in another PR - thanks.

Reply:
@Superty Fixed here: https://github.com/llvm/llvm-project/pull/128993